### PR TITLE
add templating to asset synth.py

### PR DESCRIPTION
**Pull request by crwilcox**

### CODE REVIEW

**Comment by busunkim96** [comment]
It looks like flake and black are in disagreement over some things.

**Comment by busunkim96** [comment]
We could reference Black's .flake8 configuration and update the template's .flake8 to match: https://github.com/ambv/black/blob/master/.flake8

**Comment by crwilcox** [comment]
I think the issue is I didn't commit .flake8. Adding now.

File: asset/noxfile.py [hunk 1]
```diff
@@ -20,29 +20,116 @@
...
+    # Sanity check: only run tests if found.
+    if not os.path.exists(system_test_path):
+        session.skip("System tests were not found")
+
+    # Use pre-release gRPC for system tests.
+    session.install("--pre", "grpcio")
+
+    # Install all test dependencies, then install this package into the
+    # virtualenv's dist-packages.
+    session.install("mock", "pytest")
```
Comment (tseaver) [inline asset/noxfile.py]: Hmm, shouldn't need `mock` for system tests.

**Pull request merged**